google_shopping.rb

```diff
@@ -9,7 +9,60 @@
 
 
 #outputs the full object
-puts google_data.inspect
+# puts google_data.inspect
 
 #outputs the first item's title
-puts google_data["items"][0]["product"]["title"]
\ No newline at end of file
+# puts google_data["items"][0]["product"]["title"]
+
+# outputs the number of objects
+puts "Total number of items..."
+puts google_data["items"].length
+puts ""
+
+things = google_data["items"]
+canon = []
+
+# outputs all items on backorder
+puts "Items on backorder..."
+things.each do |item|
+  if item["product"]["inventories"][0]["availability"] == "backorder"
+    puts item["product"]["title"]
+  end
+end
+puts ""
+
+# outputs all items with more than one image link
+puts "Items with more than one image link..."
+for item in things do
+  if item["product"]["images"].length > 1
+    puts item["product"]["title"]
+  end
+end
+puts ""
+
+# outputs all items that are Canon products
+puts "Items that are Canon products..."
+things.each do |item|
+  if item["product"]["brand"].downcase == "canon"
+    canon.push(item)
+    puts item["product"]["title"]
+  end
+end
+puts ""
+
+# outputs all Canon products with author name eBay
+puts "Canon products on eBay..."
+canon.each do |item|
+  if item["product"]["author"]["name"].index("eBay")
```

Comment:
This works great, another way is to use ```.include?``` to check for the presence of a substring.
```
if item["product"]["author"]["name"].downcase.include?("ebay")
```